mlflow/data/code_dataset_source.py

```diff
@@ -36,14 +38,22 @@ def _resolve(cls, raw_source: str) -> CodeDatasetSourceType:
         raise NotImplementedError
 
     def _to_dict(self) -> Dict[Any, Any]:
-        return {
-            "mlflow_source_type": self._mlflow_source_type,
-            "mlflow_source_name": self._mlflow_source_name,
-        }
+        if self._databricks_notebook_id is None:
```

Comment:
Can we also handle databricks jobs and repos? We should also store the webapp URL in case the run came from a different workspace.

For safety / completeness, can we store all the values from https://github.com/mlflow/mlflow/blob/92aa6c9575c0134e4445d289f1de3f32f73a511a/mlflow/utils/mlflow_tags.py#L30-L58? To make this easier, I think it's okay to make `CodeDatasetSource` take a collection of tags and save those tags in the JSON; no need to create a separate name for each one (e.g. `mlflow.databricks.notebookID` is fine instead of `databricks_notebook_id`)